datastream/dataset.py

```diff
@@ -89,8 +89,15 @@ def from_dataframe(dataframe: pd.DataFrame) -> Dataset[pd.Series]:
             get_item=lambda df, index: df.iloc[index],
         )
 
-    def __getitem__(self: Dataset[T], index: int) -> T:
-        '''Get an example ``T`` from the ``Dataset[T]``'''
+    def __getitem__(self: Dataset[T], index: Union[int, slice]) -> Union[T, Dataset[T]]:
+        '''Get an example or slice from the ``Dataset[T]``'''
+        if isinstance(index, slice):
+            dataframe = self.dataframe.iloc[index]
+            return Dataset(
```

Comment:
I know it's not used everywhere as it should but let's try to use `.replace` as we do in other projects

`return self.replace(dataframe=dataframe, length=len(dataframe))`